servers/src/main/java/tachyon/worker/file/FileWorkerMasterSyncExecutor.java

```diff
@@ -82,9 +82,9 @@ public void heartbeat() {
     }
     Preconditions.checkState(command.commandType == CommandType.Persist);
 
-    for (CheckpointFile checkpointFile : command.checkpointFiles) {
-      mFixedExecutionService.execute(new FilePersister(mLineageDataManager, checkpointFile.fileId,
-          checkpointFile.blockIds));
+    for (PersistFile persistFile : command.getCommandOptions().getPersistOptions().persistFiles) {
```

Comment:
It seems from the current code we want the thread to not die (debatable), so it would be more of a do nothing for this heartbeat (maybe log an error). But not something we need to address here.